src/runtime.cc

```diff
@@ -150,10 +150,14 @@ namespace detail {
 
 		m_reduction_mngr = std::make_unique<reduction_manager>();
 		m_host_object_mngr = std::make_unique<host_object_manager>();
-		m_task_mngr = std::make_unique<task_manager>(m_num_nodes, m_h_queue.get());
+		std::optional<task_recorder> t_rec;
+		if(m_cfg->is_recording()) t_rec = task_recorder{m_buffer_mngr.get()};
```

Comment:
Should this also ask for `spdlog::should_log(log_level::trace)`?

Reply:
No, the idea is that you might want to record for other reasons than writing to the log file (e.g. in a runtime unit test).